pkg/model/project.go

```diff
@@ -547,3 +547,19 @@ func (p *ProjectRBACPolicy) HasPermission(typ ProjectRBACResource_ResourceType,
 	}
 	return false
 }
+
+func VerifyRBACPermission(typ ProjectRBACResource_ResourceType, action ProjectRBACPolicy_Action, userRoles []string, roles []*ProjectRBACRole) bool {
```

Comment:
Now I understand why did you add a `Fillter...` function. 😄 
To avoid a complex parameter list of this function.

Reply:
Actually, it was. lol
I intended to refactor it but ended up with code that produces bugs.

Reply:
Just an idea, how about doing the filter inside `getRBACRoles` function and returning only the matched list?
The signature will be ` getRBACRoles(ctx context.Context, projectID string, roles []string) ([]*model.ProjectRBACRole, error)` and then we can use the returned list directly.
It is more nature. 🤔

Reply:
to reduce the complex parameter list, there is another way which is to make this verify function as closure in the `Authorize` method, then we can remove `userRoles []string & roles []*ProjectRBACRole` from the closure function signature, but it's up to you. TBH, I don't think making this VerifyRBACPermission function as part of the model package has much meaning 😄

Reply:
That's a good idea. Please apply these changes.

Reply:
That's a good idea. Please apply these changes.

Reply:
> I don't think making this VerifyRBACPermission function as part of the model package has much meaning 
I think it's better to write the verifying logic in the model pkg because `xxx.auth.go` file is automatically generated and it's not easy to change. Hence I wanna write some logic into another stuff as much as possible. wdyt?

Reply:
> xxx.auth.go file is automatically generated and it's not easy to change.

I get your point and feel the same too. But I prefer encapsulation-based codebase design, which means if the code does not actually be used in multiple places, should not move it to a common package (the model in this case). The auth codebase is autogenerated means even if we want to update it later, it should only be affected by itself template code as much as possible (reduce the risk if we change code in the common package which affect generated code).

Reply:
Oh, I see and that makes sense.
Alright, let's write `VerifyRBACPermission` in the same file.